statsmodels/genmod/generalized_linear_model.py

```diff
@@ -1040,10 +1040,11 @@ def _fit_gradient(self, start_params=None, method="newton",
         if (max_start_irls > 0) and (start_params is None):
             irls_rslt = self._fit_irls(start_params=start_params,
                                        maxiter=max_start_irls,
-                                       tol=tol, scale=scale, cov_type=cov_type,
-                                       cov_kwds=cov_kwds, use_t=use_t,
+                                       tol=tol, scale=1., cov_type='nonrobust',
+                                       cov_kwds=None, use_t=None,
                                        **kwargs)
             start_params = irls_rslt.params
+            del irls_rslt
```

Comment:
Is this specifically needed?

Reply:
It is supposed to reduce memory by deleting what is not needed anymore.

Reply:
Yah, but the gc does that too right?  The question is what is special about this object that makes it merit explicit deletion

Reply:
"It's harmless and I felt like it" is an valid answer.

Reply:
You can read the relevant issue discussion.
irls_fit is a full model/results combination that can be pretty large when nobs = 10 million.
garbage collection does not always kick in fast enough, i.e. will kick in only when irls_fit goes out of scope.
But whether it helps is difficult to tell without memory profiling.
My initial guess was that we need `start_params = irls_rslt.params.copy()` to be safer that we don't keep a reference, without having profiled that either.

"It's harmless and it most likely helps."